R-package/tests/testthat/test-RGF_package.R

```diff
@@ -288,113 +283,257 @@ testthat::test_that("the 'mat_2scipy_sparse' returns a scipy CSC sparse matrix",
   #--------------------------------------------------------------------
-  
+
   testthat::test_that("the RGF_Regressor works with sparse (scipy) matrices", {
-  
+
     skip_test_if_no_module(c("rgf.sklearn", 'scipy'))
-  
+
     set.seed(1)
     sap = sapply(1:1000, function(x) sample(c(0.0, runif(1)), 1, replace = FALSE))            # create sparse data
-  
+
     dgcM = Matrix::Matrix(data = sap,
-  
+
                           nrow = 100, ncol = 10,
-  
+
                           byrow = TRUE, sparse = TRUE)
-  
+
     scipySprse = TO_scipy_sparse(dgcM)                                            # use scipy sparse matrix
-  
+
     init_regr = RGF_Regressor$new(max_leaf = 50, sl2 = 0.1, n_iter = 10)
-  
+
     init_regr$fit(x = scipySprse, y = y_reg, sample_weight = W)                           # include also a vector of weights
-  
+
     pr = init_regr$predict(scipySprse)
-  
+
     params = unlist(init_regr$get_params(deep = TRUE))
-  
+
     validate = names(params) %in% c("normalize", "loss", "verbose", "algorithm", "n_iter", "learning_rate",
                                     "sl2", "min_samples_leaf", "opt_interval", "l2", "n_tree_search",
                                     "reg_depth", "memory_policy", "test_interval", "max_leaf")
-  
+
     tmp_score = init_regr$score(x = scipySprse, y = y_reg)
-  
+
     tmp_score_W = init_regr$score(x = scipySprse, y = y_reg, sample_weight = W)
-  
+
     testthat::expect_true( length(pr) == length(y_reg) && sum(validate) == 15 && is.double(tmp_score) && is.double(tmp_score_W) )
   })
 }
+
+
+#=========================================================================================== test feature importances
+
+
+testthat::test_that("the feature importances of the 'RGF_Regressor' class works as expected", {
+  
+  skip_test_if_no_module("rgf.sklearn")
+  
+  init_regr = RGF_Regressor$new(max_leaf = 50, sl2 = 0.1, n_iter = 10)
+  
+  init_regr$fit(x = x_rgf, y = y_reg, sample_weight = W)                    # include also a vector of weights
+  
+  vec_imp = init_regr$feature_importances()
+
+  testthat::expect_true( inherits(vec_imp, 'array') && length(vec_imp) == ncol(x_rgf) )
+})
+
+
+#=========================================================================================== test dump-model
+
+
+testthat::test_that("the 'dump_model' method returns the correct output (Dumps the forest information to the R session -- works ONLY for RGF and NOT for FastRGF)", {
+  
+  skip_test_if_no_module("rgf.sklearn")
+  
+  init_class = RGF_Classifier$new(max_leaf = 50, sl2 = 0.1, n_iter = 10)
+  
+  init_class$fit(x = x_rgf, y = y_BINclass)
+  
+  dump_model = init_class$dump_model()
+  
+  #---------------------------------------
+  # for pretty-print in the R session use:  print( dump_model() )
+  #---------------------------------------
+  
+  output_dump = reticulate::py_capture_output(dump_model())
+  
+  testthat::expect_true( nchar(output_dump) > 0 && object.size(output_dump) > 0 )
```

Comment:
@StrikerRUS, I searched in the python tests but I don't see any corresponding test for the 'dump_model()' function. Would you mind adding one so that I have a guideline to implement it in R? thanks.

Reply:
Refer to
https://github.com/RGF-team/rgf/blob/44b3654b5801273bc1a2e0d291a734f43d6b0b48/python-package/rgf/rgf_model.py#L403-L415

I meant something that in Python looks like `output_dump.find("[  0], depth=0, gain=") != -1`. It will allow to be sure that dumped output contains a real model dump but not any error message, for example.

Reply:
Done!